omnn/math/Valuable.h

```diff
@@ -658,14 +666,25 @@ class Valuable
 
     [[nodiscard]] virtual bool is_optimized() const;
 
-protected:
+private:
+    friend class boost::serialization::access;
+
     View view = View::Flat;
-    bool optimized = false;
-    void MarkAsOptimized();
 
-    //   TODO : std::shared_ptr<std::vector<Valuable>> cachedValues;
+    template<class Archive>
+    void serialize(Archive & ar, const unsigned int version) {
+        ar & exp;
```

Comment:
rather exp->serialize(...) but it requires polymorphic call, so you need a virtual serialize method which accepts stream interface

Reply:
There is no virtual template methods feature in C++ yet.
Lambda can accept auto params which is the same as templates. 
This approach may be used to workaround the polymorphic type serialization.